### PR TITLE
Use prettier

### PR DESCRIPTION
**Pull request by hunger**
Simon suggested to do this in #1506:

* Add a prettier config file with 4 spaces per tab
* Update the .pre-commit.yaml (updated and prittified) -- this insists to be on a separate commit:-)
* Update all other files
* Add a CI job to verify that everything is pretty :-)

One way to make sure everything is fine before pushing is of course [pre-commit](https://pre-commit.com/), which has a rule to run prettier (and a lot more like cspell) on git commit.

### CODE REVIEW

File: docs/resources/slint-docs-highlight.html [hunk 1]
```diff
@@ -3,101 +3,48 @@
     It can be injected via the `--html-in-header slint-docs-highlight.html` option of rustdoc, is included via _templates/layout.html
     in sphinx and via head.hbs in mdbook.
 -->
-<link rel="stylesheet" href="https://slint-ui.com/resources/highlightjs/11.0.1/default.min.css">
+<link
+    rel="stylesheet"
+    href="https://slint-ui.com/resources/highlightjs/11.0.1/default.min.css"
+/>
```
Comment (ogoffart) [inline docs/resources/slint-docs-highlight.html:9]: Why does it split this?

File: docs/testing.md [hunk 2]
```diff
@@ -61,7 +60,7 @@ Foo := Rectangle {
 ### Rust driver
 
 The rust driver will compile each snippet of code and put it in a `slint!` macro in its own module
-In addition, if there are ```` ```rust ```` blocks in a comment, they are extracted into a `#[test]`
+In addition, if there are ` ```rust ` blocks in a comment, they are extracted into a `#[test]`
```
Comment (ogoffart) [inline docs/testing.md:63]: This looks wrong

File: docs/resources/slint-docs-highlight.html [hunk 3]
```diff
@@ -3,101 +3,48 @@
...
+    (slintBlock of document.querySelectorAll( "[class*=language-slint]" )) {
+    hljs.highlightElement(slintBlock); } // Some of the rustdoc selectors
+    require the pre element to have the rust class for (codeBlock of
+    document.querySelectorAll( ".language-slint.hljs" )) {
+    codeBlock.parentElement.classList.add("rust"); } // Change the hljs
+    generated classes to the rustdoc // ones, so that the highlighting adjusts
+    to the theme correctly. const highlightJSToRustDoc = [ ["comment",
+    "comment"], ["number", "number"], ["symbol", "struct"], // width:
+    ["keyword", "kw"], ["built_in", "primitive"], ["string", "string"],
+    ["title", "fnname"], // Foo := ["type", "type"], ]; for ([hljs_class,
```
Comment (ogoffart) [inline docs/resources/slint-docs-highlight.html:40]: that's wrong

**Review by ogoffart — COMMENTED** [review]
a few things went wrong in there or are not really pretty.

TBH, i'm not a fan.

File: docs/resources/slint-docs-highlight.html [hunk 4]
```diff
@@ -3,101 +3,48 @@
...
-      hljs.highlightAll();
-
-      // The Sphinx/my_st generated HTML for code blocks does not use <code> tags, so highlight.js'
-      // default selector "pre code" does not match. Let's do it by hand:
-      for (block of document.querySelectorAll("div[class*=highlight-slint] div.highlight pre")) {
-        hljs.highlightElement(block)
-      }
-    }
-  });
+    function highlight_slint(hljs) { const KEYWORDS = { keyword: "struct export
```
[the remark this answers is not in the document]
Comment (hunger) [reply docs/resources/slint-docs-highlight.html:12]: Worse: It's broken!

File: docs/tutorial/rust/src/ideas_for_the_reader.md [hunk 5]
```diff
@@ -2,13 +2,13 @@
 
 The game is visually a little bare. Here are some ideas how you could make further changes to enhance it:
 
-* The tiles could have rounded corners, to look a little less sharp. The [border-radius](https://slint-ui.com/docs/rust/slint/docs/builtin_elements/index.html#rectangle)
-  property of *Rectangle* can be used to achieve that.
+-   The tiles could have rounded corners, to look a little less sharp. The [border-radius](https://slint-ui.com/docs/rust/slint/docs/builtin_elements/index.html#rectangle)
+    property of _Rectangle_ can be used to achieve that.
```
Comment (tronical) [inline docs/tutorial/rust/src/ideas_for_the_reader.md:6]: Perhaps there's a way to configure prettier not to do this change and also make it less noisy? It seems to me that underscore is not recommended: https://www.markdownguide.org/basic-syntax/#italic-best-practices

**Comment by ogoffart** [comment]
There is a few things in this formatting style i really don't like:
 - in bullet list, why so much space between the `-` and the item
 - I like to be able to freely choose between to use `-` or `*` for bullets list
 - In HTML, i don't like the tags being split after the `<`

Looks like prettier is not easy to configure (at least they claim it's opinionated and meant like that) and I don't think it's worth the trouble.
I'd rather you configure prettier to ignore markdown and HTML or remove it from your precommit hook.

**Comment by hunger** [comment]
This patch is controversial and does break some things, so let's not go through with it!

**Pull request closed without merging**